### DISCUSSION TITLE
Realtek RTL8822CE on Lenovo V15 G2 ALC

### DISCUSSION THREAD

**Discussion by windigerbus** [vanilla-wiiu/vanilla#84] [Hardware Results]
This adapter actually works fine, the cause of the error in the original post was a broken FFmpeg install.
Original post:

> Video doesn't work, connecting for the first time it's frozen on the first frame and reconnecting I get a black screen. The current wiki entry says that this Wi-Fi adapter should just work. Input and audio is working.
> Here's some info:
> ```
> 02:00.0 Network controller: Realtek Semiconductor Co., Ltd. RTL8822CE 802.11ac PCIe Wireless Network Adapter
> ```
> ```
> OS: Fedora Linux 40 (Workstation Edition) x86_64
> Host: 82KD (Lenovo V15 G2 ALC)
> Kernel: Linux 6.11.3-200.fc40.x86_64
> DE: GNOME 46.5
> CPU: AMD Ryzen 5 5500U (12) @ 4.06 GHz
> GPU: AMD Lucienne [Integrated]
> ```
> It's spamming this error:
> ```
> [libopenh264 @ 0x35be4d80] DecodeFrame failed
> Failed to send packet to decoder: -1313558101
> ```

**Comment by karmatic-valkyrie** [comment]
I have been using a Lenovo V14 G4 ABP - Type 83FG with a Realtek RTL8822CE. My laptop seems to have pretty much the exact same hardware as OP's, with the same CPU and integrated GPU, the same amount (8 GB) of 3200 MHz DDR4 RAM, and probably even the same 256 GB SSD. The only major differences are that I'm using Arch Linux as my OS and KDE Plasma as my DE. Vanilla has been able to connect to my Wii U without any significant issues for the past several months that I have been using it, both with the main software and the pipe. The audio, video, and input all work and apart from some very minor video and audio delay, it has been nearly as good as a real Wii U Gamepad for me.

~~Granted, I started using Vanilla more than a few months after this post was made and it has now been almost a full year. The project has seen some major changes since, so maybe the issues OP mentioned above were fixed in that time period and/or weren't related to the compatibility of the WiFi card itself? I really don't know.~~

Edit: I'm just speculating based upon what I saw posted online about the same or very similar errors, but it doesn't seem like it's an issue with the WiFi card and it's probably not even an issue caused by Vanilla itself. I think the error listed was caused by a memory issue for the software decoder itself. It's either an issue caused by FFmpeg, libopenh264, or bad blocks in the RAM itself. ~~It doesn't really matter at this point though, seeing as OP never made any subsequent comments or bug reports in this repository over the past year and probably doesn't care or resolved it on his own.~~

Anyways, sorry for necrobumping this thread. I just felt that it was better to post my experience here with the Realtek RTL8822CE rather than creating a whole new thread, especially considering the fact that I'm using nearly the exact same hardware described here.

**Reply by windigerbus** [reply]
> I have been using a Lenovo V14 G4 ABP - Type 83FG with a Realtek RTL8822CE. My laptop seems to have pretty much the exact same hardware as OP's, with the same CPU and integrated GPU, the same amount (8 GB) of 3200 MHz DDR4 RAM, and probably even the same 256 GB SSD. The only major differences are that I'm using Arch Linux as my OS and KDE Plasma as my DE. Vanilla has been able to connect to my Wii U without any significant issues for the past several months that I have been using it, both with the main software and the pipe. The audio, video, and input all work and apart from some very minor video and audio delay it's as good as a real Wii U Gamepad.
> 
> ~Granted, I started using Vanilla more than a few months after this post was made and it has now been almost a full year. The project has seen some major changes since, so maybe the issues OP mentioned above were fixed in that time period and/or weren't related to the compatibility of the WiFi card itself? I really don't know.~
> 
> Edit: I'm just speculating based upon what I saw posted online about the same or very similar errors, but it doesn't seem like it's an issue with the WiFi card and it's probably not even an issue caused by Vanilla itself. I think the error listed was caused by a memory issue for the software decoder itself. It's either an issue caused by FFmpeg, libopenh264, or bad blocks in the RAM itself. It doesn't really matter at this point though, seeing as OP never made any subsequent comments or bug reports in this repository over the past year and probably doesn't care or resolved it on his own.
> 
> Anyways, sorry for necrobumping this thread. I just felt that it was better to post my experience here with the Realtek RTL8822CE rather than creating a whole new thread, especially considering the fact that I'm using nearly the exact same hardware described here.

I forgot about this post, it was a combination of my broken ffmpeg install and my firewall configuration! It does work well now.

**Reply by karmatic-valkyrie** [reply]
It's cool to hear that you managed to fix it and that it's working for you! Also, I encountered similar issues with my firewall configuration when I was using Vanilla initially as well.  It seems like a somewhat common issue to run into for new users.
